server/models/snippet.js

```diff
@@ -1,30 +1,62 @@
-const mongoose = require("mongoose");
+const mongoose = require('mongoose');
 
 const snippetSchema = new mongoose.Schema({
-	code: {
-		type: String,
-		required: true,
-		
-	},
-	author: {
-		type: mongoose.Schema.Types.ObjectId,
-		required: true,
-		ref: "User",
-	},
-	reviewer: {
-		type: mongoose.Schema.Types.ObjectId,
-		ref: "User",
-	},
-	title: {
-		type: String,
-		required: true,
-		default: "Unknown",
-	},
-	language: {
-		type: String,
-		required: true,
-	},
+  code: {
+    type: String,
+    required: true,
+  },
+  type: {
+    type: String,
+    required: true,
+    enum: ['request', 'receive'],
+  },
+  status: {
+    type: String,
+    enum: ['pending', 'accepted', 'completed'],
+    default: 'pending',
+  },
+  author: {
+    type: mongoose.Schema.Types.ObjectId,
+    required: true,
+    ref: 'User',
+  },
+  reviewer: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'User',
+  },
+  title: {
+    type: String,
+    required: true,
+    default: 'Unknown',
+  },
+  language: {
+    type: String,
+    required: true,
+  },
+  comments: {
+    type: String,
+    required: true,
+  },
+  rating: {
+    type: Number,
+    min: 0,
+    max: 5,
+  },
+  date_requested: {
+    type: Date,
+    required: true,
+    default: Date.now,
+  },
+  date_submitted: {
+    type: Date,
+    required: true,
+    default: Date.now,
+  },
+  date_accepted: {
+    type: Date,
+    required: true,
+  },
 });
 
-const Snippet = mongoose.model("Snippet", snippetSchema);
+const Snippet = mongoose.model('Snippet', snippetSchema);
```

Comment:
I think we should set a rule to use `'` or `"` . I personally prefer `'` . But I reconfigured my prettier formatter to use `"` because it was like that in the template we started the project on. It's not a big deal generally, but now we get merge conflicts only because of `'` and `"`  :D 
What do you think @aeciorc ?

Reply:
I think it's not a problem. The app still works with both. When you clone it, for instance, depending on the prettier format, it will format it for us. I prefer single quote as well. But I can set to double quote for later on.

Reply:
Yeah it does works. But it causes merge conflicts for me. Doesn't it for you?

Reply:
I have no problem with it so far.